client_push_pre_go17.go

```diff
@@ -0,0 +1,53 @@
+// +build !go1.7
+
+package apns2
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// Push sends a Notification to the APNs gateway. If the underlying http.Client
+// is not currently connected, this method will attempt to reconnect
+// transparently before sending the notification. It will return a Response
+// indicating whether the notification was accepted or rejected by the APNs
+// gateway, or an error if something goes wrong.
+//
+// Context with deadline allows to close long request when context timeout exceed. It can be nil, for back compatibility
+func (c *Client) PushWithCtx(n *Notification, ctx context.Context) (*Response, error) {
```

Comment:
The problem with the duplication is that future changes need to made in both places. If they are not, the code could diverge in behaviour between Go versions. If there isn't any other option, good tests and continuous integration on multiple Go versions can help.

I await a verdict/suggestions from @sideshow.